core/execution/engine.go

```diff
@@ -983,6 +986,11 @@ func (e *Engine) SubmitStopOrders(
 	}
 
 	if mkt, ok := e.allMarkets[market]; ok {
+		// Stop orders are rejected if the market is in the opening auction
+		if mkt.IsOpeningAuction() {
+			return nil, ErrStopOrdersNotAcceptedDuringOpeningAuction
```

Comment:
Would it not be better to add a rejection reason in the enum, and reject it from inside the market like other rejections so an event is send and all that stuff?

Reply:
Is that better?